app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallScreen.kt

```diff
@@ -44,36 +43,43 @@ import com.wire.android.ui.theme.wireDimensions
 @Composable
 fun OngoingCallScreen(
     lifecycleOwner: LifecycleOwner = LocalLifecycleOwner.current,
-    ongoingCallViewModel: OngoingCallViewModel = hiltViewModel()
+    ongoingCallViewModel: OngoingCallViewModel = hiltViewModel(),
+    sharedCallingViewModel: SharedCallingViewModel = hiltViewModel()
 ) {
-    OngoingCallContent(ongoingCallViewModel)
-    observeScreenLifecycleChanges(lifecycleOwner, ongoingCallViewModel)
+    OngoingCallContent(sharedCallingViewModel)
+    observeScreenLifecycleChanges(lifecycleOwner, sharedCallingViewModel)
 }
 
 @OptIn(ExperimentalMaterialApi::class)
 @Composable
-private fun OngoingCallContent(ongoingCallViewModel: OngoingCallViewModel) {
+private fun OngoingCallContent(
+    sharedCallingViewModel: SharedCallingViewModel
+) {
 
     val scaffoldState = rememberBottomSheetScaffoldState()
     BottomSheetScaffold(
         topBar = {
-            //TODO to handle null name in different way
             OngoingCallTopBar(
-                ongoingCallViewModel.callEstablishedState.conversationName
-                    ?: stringResource(id = R.string.calling_label_default_caller_name)
-            ) {}
+                conversationName = when (sharedCallingViewModel.callState.conversationName) {
+                    is ConversationName.Known -> (sharedCallingViewModel.callState.conversationName as ConversationName.Known).name
```

Comment:
this cast should not be necessary I think ?

Reply:
The compiler is raising some errors there, I will do it in a different way